db/db_impl.cc

```diff
@@ -1085,7 +1085,7 @@ std::vector<Status> DBImpl::MultiGet(
         read_options.snapshot)->number_;
   } else {
     snapshot = allocate_seq_only_for_data_ ? versions_->LastSequence()
-                                           : versions_->LastAllocatedSequence();
+                                           : versions_->LastPublishedSequence();
```

Comment:
have you update NewIterator and NewIterators?

Reply:
if you mean in db_impl, we already have this comment to avoid confusion: https://github.com/facebook/rocksdb/blob/master/db/db_impl.cc#L1495
If you mean WritePreparedTxnDB::NewIterator, the use GetSnapshot, which already includes this update.